okhttp/src/main/java/okhttp3/internal/http2/Header.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.http2
+
+import okhttp3.internal.Util
+import okio.ByteString
+
+/** HTTP header: the name is an ASCII string, but the value can be UTF-8. */
+class Header(
+  /** Name in case-insensitive ASCII encoding. */
+  @JvmField val name: ByteString,
+  /** Value in UTF-8 encoding. */
+  @JvmField val value: ByteString
+) {
+  @JvmField internal val hpackSize = 32 + name.size() + value.size()
+
+  // TODO: search for toLowerCase and consider moving logic here.
+  constructor(name: String, value: String) : this(ByteString.encodeUtf8(name), ByteString.encodeUtf8(value))
+
+  constructor(name: ByteString, value: String) : this(name, ByteString.encodeUtf8(value))
+
+  override fun equals(other: Any?): Boolean {
+    return other is Header
+        && this.name == other.name
+        && this.value == other.value
+  }
+
+  override fun hashCode(): Int {
+    var result = 17
+    result = 31 * result + name.hashCode()
+    result = 31 * result + value.hashCode()
+    return result
+  }
+
+  override fun toString(): String {
+    return Util.format("%s: %s", name.utf8(), value.utf8())
```

Comment:
Use interpolation directly?

Reply:
`Util.format` formats the string according to US locale. Just wanted to confirm that moving to direct interpolation won't affect that? (unsure about locale implications with direct interpolation).

Reply:
It's relevant for `%d` but not `%s`.
https://github.com/square/okhttp/issues/2522